### PR TITLE
[WIP] feat(server): support ssr running in vm

### PR DESCRIPTION
**Pull request by Amour1688**
### <samp>🤖 Generated by Copilot at 1d351d4</samp>

Refactored and improved server-side rendering logic for prod-server. Extracted rendering code to `runServerRender` function in `libs/render/ssr/utils.ts` and added VM mode support for better isolation and security. Moved `utils.ts` to `ssr` folder for better organization.

### <samp>🤖 Generated by Copilot at 1d351d4</samp>

*  Simplify and enable VM mode for server-side rendering logic ([link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-11c102982d4c6e9170690b8a835eabccf0e03018adbf774498acb0ce7c97047fL7-R16), [link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-11c102982d4c6e9170690b8a835eabccf0e03018adbf774498acb0ce7c97047fL86-R93), [link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-cc2e8c4aa8620bef02b20fc89fcf538dd685dcfe0f202b910d6b0824ccf32bf2L6-R13), [link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-cc2e8c4aa8620bef02b20fc89fcf538dd685dcfe0f202b910d6b0824ccf32bf2R41-R83))
  * Remove unused `SERVER_RENDER_FUNCTION_NAME` import from `ssr.ts` and add `runServerRender` import from `utils.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-11c102982d4c6e9170690b8a835eabccf0e03018adbf774498acb0ce7c97047fL7-R16))
  * Replace bundleJS require and serverRender call with `runServerRender` function in `render` function in `ssr.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-11c102982d4c6e9170690b8a835eabccf0e03018adbf774498acb0ce7c97047fL86-R93))
  * Move `utils.ts` file from `libs/render` to `libs/render/ssr` and add or remove imports accordingly ([link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-cc2e8c4aa8620bef02b20fc89fcf538dd685dcfe0f202b910d6b0824ccf32bf2L6-R13))
  * Add `runServerRender` function to `utils.ts` to handle the server-side rendering logic with or without VM mode ([link](https://github.com/web-infra-dev/modern.js/pull/4420/files?diff=unified&w=0#diff-cc2e8c4aa8620bef02b20fc89fcf538dd685dcfe0f202b910d6b0824ccf32bf2R41-R83))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by zoolsher** [comment]
Do you mean a natural raw JavaScript Context without no API support by "VM"?

File: packages/server/prod-server/src/libs/render/utils.ts [hunk 1]
```diff
@@ -33,3 +38,46 @@ export const injectServerDataStream = (
...
+      module: {},
+    };
+    runInNewContext(
+      `
+${await fs.readFile(bundleJS, 'utf-8')}
+globalThis.__SERVER_RENDER__ = module.exports.serverRender;
+    `,
+      global,
+    );
+    return global.__SERVER_RENDER__(serverContext);
```
Comment (spencerHT) [inline packages/server/prod-server/src/libs/render/utils.ts:78]: Should this content need to cached with `cache` function?

Comment (spencerHT) [reply packages/server/prod-server/src/libs/render/utils.ts:78]: And I think `global.__SERVER_RENDER__` should use `runInNewContext` for finally running `renderToString` in new global.

**Review by GiveMe-A-Name — APPROVED** [review]

**Comment by chenjiahan** [comment]
Closed as the PR is outdated, feel free to reopen it if it is still needed.

**Pull request closed without merging**